### PR TITLE
Moma analyzer improvements

### PR DESCRIPTION
**Pull request by cristianst85**
Properly parse type for methods with parameters that supports generics.
Added more type aliases conversions.
Added test cases.

**Before**
bug-malformed-method-signature-before

**After**
bug-malformed-method-signature-after

### CODE REVIEW

**Comment by marek-safar** [comment]
Please don't change existing formatting

**Comment by cristianst85** [comment]
Can you be more specific please?

Sent from my Windows 10 phone

File: MoMA.Analyzer/Methods/Method.cs [hunk 1]
```diff
@@ -151,28 +151,48 @@ public string ToStringWithClass ()
 		
 		private string ConvertType (string type)
 		{
-			switch (type) {
+            switch (type) {
```
Comment (marek-safar) [inline MoMA.Analyzer/Methods/Method.cs]: Please use same formatting

**Comment by cristianst85** [comment]
The original file is badly formatted. If can switch to using tabs, but there are a lot of changes then.

File: MoMA.Analyzer/Methods/Method.cs (hunk 1 above)
Comment (cristianst85) [inline MoMA.Analyzer/Methods/Method.cs]: The original file is badly formatted. I can switch to using tabs, but then there are a lot of changes.

**Comment by marek-safar** [comment]
Please rebase on top of master

**Comment by cristianst85** [comment]
I've removed this because I don't have time for this bs. Please fix your own repo first (add proper .gitignore file, etc.)

**Pull request closed without merging**